tardis/io/util.py

```diff
@@ -288,6 +289,27 @@ def from_hdf_util(cls, file_path, path=''):
                             hdf[key] = None
         return hdf
 
+    @classmethod
+    def from_hdf(cls, file_path, path='', name=None):
+
+        if name is None:
+            name = cls.__name__
+
+        buff_path = os.path.join(path, name)
+        data = cls.from_hdf_util(file_path, buff_path)
+        for key, value in cls.class_properties.items():
+            data[key] = value.from_hdf(file_path, buff_path, name=key)
+
+        #Get initialization paramaters from constructor definition of class
+        argspec = inspect.getargspec(cls.__init__).args
+        argspec.remove('self')
+
+        initializer_dict = {name: data[name] for name in argspec}
+
+        try:
+            return cls(**initializer_dict)
+        except:
+            return data
```

Comment:
I think `from_hdf` should always return an instance of class `cls`. If that is not possible (missing data), then we should raise an error detailing which values were missing.